src/process.rs

```diff
@@ -505,6 +526,52 @@ async fn process_cloudtrail(
     }
 }
 
+pub async fn kafka_logs(
+    topic: String,
+    records:  Vec<KafkaRecord>,
+    coralogix_exporter: DynLogExporter,
+    config: &Config,
+) -> Result<(), Error> {
+    let defined_app_name = config
+        .app_name
+        .clone()
+        .unwrap_or_else(|| "NO APPLICATION NAME".to_string());
+
+    let defined_sub_name = config
+        .sub_name
+        .clone()
+        .unwrap_or(topic);
+
+    let mut batch = Vec::new();
+    for record in records {
+        if record.value.is_none() {
+            continue
+        }
+
+        // check if value is base64 encoded
+        if let Ok(message) = BASE64_STANDARD.decode(record.value.clone().unwrap()) {
+            batch.push(String::from_utf8(message)?);
+            continue
+        }
+
+        batch.push(record.value.unwrap());
```

Comment:
It would be better to avoid unwraps and clone:
```suggestion
        if let Some(value) = record.value {
            // check if value is base64 encoded
            if let Ok(message) = BASE64_STANDARD.decode(&value) {
                batch.push(String::from_utf8(message)?);
            } else {
                batch.push(value);
            }
        }
```

Reply:
@rtimush , there's a check to insure the value isn't None on line 547, so technically the unwrap should never panic, however, i see your point.